packages/application-to-register-webapp/src/routes/__tests__/land/legal-agreement-start-date-spec.js

```diff
@@ -0,0 +1,126 @@
+
+      const response = await submitPostRequest(postOptions)
+      expect(response.statusCode).toBe(302)
+    })
+
+    it('should fail to add a legal agreement start date with empty dates', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = ''
+      postOptions.payload['legalAgreementStartDate-month'] = ''
+      postOptions.payload['legalAgreementStartDate-year'] = ''
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add a legal agreement start date with empty day', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = ''
+      postOptions.payload['legalAgreementStartDate-month'] = '01'
+      postOptions.payload['legalAgreementStartDate-year'] = '2022'
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add a legal agreement start date with empty month', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = '01'
+      postOptions.payload['legalAgreementStartDate-month'] = ''
+      postOptions.payload['legalAgreementStartDate-year'] = '2022'
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add a legal agreement start date with empty year', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = '01'
+      postOptions.payload['legalAgreementStartDate-month'] = '01'
+      postOptions.payload['legalAgreementStartDate-year'] = ''
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add a legal agreement start date with invalid dates', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = 'xx'
+      postOptions.payload['legalAgreementStartDate-month'] = 'zz'
+      postOptions.payload['legalAgreementStartDate-year'] = 'cc'
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add a legal agreement start date with wrong day', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = 'x1'
+      postOptions.payload['legalAgreementStartDate-month'] = '11'
+      postOptions.payload['legalAgreementStartDate-year'] = '2020'
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add a legal agreement start date with wrong month', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = '11'
+      postOptions.payload['legalAgreementStartDate-month'] = '23'
+      postOptions.payload['legalAgreementStartDate-year'] = '2020'
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
```

Comment:
Please can you add additional checks for expected errors. [For example](https://github.com/DEFRA/biodiversity-net-gain-service/blob/master/packages/application-to-register-webapp/src/routes/__tests__/land/add-landowners.spec.js).